tests/lwip_sock_dns/tests/01-run.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2022 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from time import sleep
+from testrunner import run
+
+
+def _test_dns_requests(child):
+    child.sendline('dns request 0 example.com')
+    child.expect_exact('example.com resolves to 2606:2800:220:1:248:1893:25c8:1946')
+
+    child.sendline('dns request 4 example.com')
+    child.expect_exact('example.com resolves to 93.184.216.34')
+
+    child.sendline('dns request 6 example.com')
+    child.expect_exact('example.com resolves to 2606:2800:220:1:248:1893:25c8:1946')
+
+
+def test_no_dns_server(child):
+    child.sendline('dns server')
+    child.expect_exact('DNS server: -')
+
+
+def test_invalid_ipv4(child):
+    child.sendline('dns server 4 1.2.3.4.5')
+    child.expect_exact('invalid IPv4 address')
+
+
+def test_invalid_ipv6(child):
+    child.sendline('dns server 6 1:2:3:4:5:6:7:8:9')
+    child.expect_exact('invalid IPv6 address')
+
+
+def test_ipv4_dns_server(child):
+    child.sendline('dns server 4 8.8.8.8')
+    child.sendline('dns server')
+    child.expect_exact('DNS server: 8.8.8.8:53')
+    _test_dns_requests(child)
+
+
+def test_ipv6_dns_server(child):
+    child.sendline('dns server 6 2001:4860:4860::8888')
+    child.sendline('dns server')
+    child.expect_exact('DNS server: [2001:4860:4860::8888]:53')
+    _test_dns_requests(child)
```

Comment:
https://developers.google.com/speed/public-dns/docs/using

They are probably more stable than the example domain data.

Reply:
Wouldn't be the first service Google shuts down because they feel like it (even though people rely on it)... but I am not blocking because of that. I would prefer a setup where a server is provided (using scapy e.g.), so at some point it _could_ be run automatically, but I won't block the PR because of that.

Reply:
I agree that this would be nice, but for me (personally) it would be simply an overkill for what this test is for me: a by-product of playing around with lwIP ;) From the available DNS servers like Quad9 or Cloudflare I would say that I trust Google the most when it comes to "How long will this service still be available".